### PR TITLE
Fixed up single letter vars in Fire Alarm code and introduces a small feature

### PR DESCRIPTION
**Pull request by DuffCreeper**
## About The Pull Request

Cleaned up firealarm.dm to give single letter vars a name while also adding a new feature that lets people examine the working state of the thermal sensor. Silicons can disable/enable said sensor using a key shortcut

## Why It's Good For The Game

I'm not quite sure what to say for the examine text part of this. I only thought about it when I wanted to allow silicons to enable/disable the sensors to 'balance' it out. Mainly due to the part if a shit ton of sensors are disabled, something aint quite right. Anyway that isn't exactly the reason why I added this. It's more to help with trying to equalize temperatures between rooms if a bit of cold is tripping a fire alarm and you want to use heat from other rooms to basically bring the temp up, of course fire alarms can be.. counter productive on maps where a fire alarm panel is in a pocket of fire doors with no vent or anything to bring the temperature up

TL:DR I just want it to be easier to control low temperatures and fix the issue of a lone fire alarm constantly tripping due to there being no vent/scrubber where it is

## Changelog
:cl:
add: Allows people to now examine fire alarms to detect their thermal sensor state. Silicons can now also Ctrl+Click to enable and disable at the sensor at a whim
code: Cleaned up single letter variables in firealarm.dm
/:cl:

### CODE REVIEW

**Comment by Nistishere** [comment]
QoL change, overrides Thermal sensor Override from malf AI entirely.

**Comment by DuffCreeper** [comment]
> QoL change, overrides Thermal sensor Override from malf AI entirely.

I've honestly forgotten that was a thing malf can get

**Comment by Nistishere** [comment]
> > QoL change, overrides Thermal sensor Override from malf AI entirely.
> 
> I've honestly forgotten that was a thing malf can get

Honestly fair. Still like the change tbh. Frees space for better malf modules in the future

**Comment by LemonInTheDark** [comment]
```
Allows people to now examine fire alarms to detect their thermal sensor state. Silicons can now also Ctrl+Click to enable and disable at the sensor at a whim
```
There is a way for borgs to disable fire alarms, but it requires some knowledge (God I need to update the wiki), and the engi module. Is this change directed at the ai? or silicons in general.

On an entirely unrelated note, thank you for cleaning the single letters up, that's really nice to see

**Comment by Nistishere** [comment]
> There is a way for borgs to disable fire alarms, but it requires some knowledge

Yep. Screwdriver-multitool.

**Comment by DuffCreeper** [comment]
> > Allows people to now examine fire alarms to detect their thermal sensor state. Silicons can now also Ctrl+Click to enable and disable at the sensor at a whim
> 
> There is a way for borgs to disable fire alarms, but it requires some knowledge (God I need to update the wiki), and the engi module. Is this change directed at the ai? or silicons in general.
> 
> On an entirely unrelated note, thank you for cleaning the single letters up, that's really nice to see

I'd say in general, borgs and AI can both interact with the fire alarm from afar so I dont see why only the AI can. Fair game to give it to both imo. But that's just me and how I view the practical uses for them

File: code/game/machinery/firealarm.dm [hunk 1]
```diff
@@ -218,32 +218,32 @@
...
+				else if(tool.force) //hit and turn it on
 					..()
-					var/area/A = get_area(src)
-					if(!A.fire)
+					var/area/area = get_area(src)
+					if(!area.fire)
 						alarm()
 					return
 
 			if(1)
```
Comment (ZephyrTFA) [inline code/game/machinery/firealarm.dm:244]: if(1)???

Comment (ZephyrTFA) [reply code/game/machinery/firealarm.dm:244]: Why are we using magic numbers 😭

Comment (DuffCreeper) [reply code/game/machinery/firealarm.dm:244]: cause im scared to touch build stages and dunno what to name them to 😔
i am big brain

File: code/game/machinery/firealarm.dm [hunk 2]
```diff
@@ -348,6 +348,25 @@
...
+/obj/machinery/firealarm/examine(mob/user)
+	. = ..()
+	. += "A light on the side indicates the thermal sensor is [detecting ? "enabled" : "disabled"]."
+
+// Allows Silicons to disable thermal sensor
+/obj/machinery/firealarm/BorgCtrlClick(mob/living/silicon/robot/user)
+	if(get_dist(src,user) <= user.interaction_range)
+		AICtrlClick(user)
+	else
+		..()
```
Comment (Rohesie) [inline code/game/machinery/firealarm.dm:361]:
```suggestion
/obj/machinery/firealarm/BorgCtrlClick(mob/living/silicon/robot/user)
	if(get_dist(src,user) <= user.interaction_range)
		AICtrlClick(user)
		return
	return ..()
```
* Preserve parent returns unless you have specific reasons not to.
* Make use and abuse of early returns to diminish horizontal drift.

Comment (DuffCreeper) [reply code/game/machinery/firealarm.dm:361]: Huh gotcha shall keep that in mind for the future

**Comment by Rohesie** [comment]
Should be good after the merge conflict is fixed.

**Comment by DuffCreeper** [comment]
> Should be good after the merge conflict is fixed.

I half expected to fuck up the merge conflict by the fact I keep pressing P instead of O. Glad to see the linters passed

File: code/game/machinery/firealarm.dm [hunk 3]
```diff
@@ -348,6 +348,25 @@
...
+// Allows Silicons to disable thermal sensor
+/obj/machinery/firealarm/BorgCtrlClick(mob/living/silicon/robot/user)
+	if(get_dist(src,user) <= user.interaction_range)
+		AICtrlClick(user)
+		return
+	return ..()
+
+/obj/machinery/firealarm/AICtrlClick(mob/living/silicon/robot/user)
+	if(obj_flags & EMAGGED)
+		to_chat(user, "<span class='warning'>The control circuitry of [src] appears to be malfunctioning.</span>")
```
Comment (Rohesie) [inline code/game/machinery/firealarm.dm:365]: Use the new procs for the span classes now, please.

File: code/game/machinery/firealarm.dm [hunk 4]
```diff
@@ -348,6 +348,25 @@
...
+		return
+	detecting = !detecting
+	to_chat(user, "<span class='notice'>You [ detecting ? "enable" : "disable" ] [src]'s detecting unit!</span>")
```
Comment (Rohesie) [inline code/game/machinery/firealarm.dm:368]: Same here.

File: code/game/machinery/firealarm.dm (hunk 3 above)
Comment (DuffCreeper) [reply code/game/machinery/firealarm.dm:365]: I'll fix them up when I wake up then. Thanks for pointing them out
